examples/gcf-pubsub-vm-delete-event-handler/main.py

```diff
@@ -0,0 +1,660 @@
+    events.  Primarily configuration, network API clients, and logging API
+    clients.
+    """
+    LOGNAME = 'dns-vm-gc'
+
+    def __init__(self, http=None, session=None):
+        """Initializes the app to handle multiple events
+
+        Args:
+            http: httplib2.Http, An instance of httplib2.Http or something that
+                acts like it that HTTP requests will be made through.
+            session: A requests.Session instance intended for mocking out the
+                Stackdriver API when under test.
+        """
+        # Log clients
+        self.log = self.setup_python_logging()
+        self.cloud_log = self.setup_cloud_logging(session=session)
+        # API clients
+        self.compute = discovery.build('compute', 'v1', http=http)
+        self.dns = discovery.build('dns', 'v1', http=http)
+
+    def setup_python_logging(self):
+        """Configures Python logging system
+
+        Python logs are sent to STDOUT and STDERR by default.  In GCF, these
+        logs are associated on execution_id.
+        """
+        if os.getenv('DEBUG'):
+            level = logging.DEBUG
+        else:
+            level = logging.INFO
+        # Make googleapiclient less noisy.
+        # See https://github.com/googleapis/google-api-python-client/issues/299
+        api_logger = logging.getLogger('googleapiclient')
+        api_logger.setLevel(logging.ERROR)
+        # Set level of our logger.
+        log = logging.getLogger(self.LOGNAME)
+        log.setLevel(level)
+        return log
+
+    def setup_cloud_logging(self, session=None):
+        """Configures Structured Logging for results reporting
+
+        Structured logs are used to report the results of execution.  This is
+        different from Python logging used to report step by step progress of a
+        single execution.
+
+        Args:
+            session: A requests.Session instance intended for mocking out the
+                Stackdriver API when under test.
+        """
+        if session:
+            client = google.cloud.logging.Client(
+                _http=session,
+                _use_grpc=False
+            )
+        else:
+            client = google.cloud.logging.Client()
+        logger = client.logger(self.LOGNAME)
+        return logger
+
+    def handle_event(self, data, context=None):
+        """Background Cloud Function to delete DNS A records when VM is deleted.
+
+        Args:
+            data (dict): The dictionary with data specific to this type of
+                event.
+            context (google.cloud.functions.Context): The Cloud Functions event
+                metadata.
+        Returns:
+            Number of records deleted across all managed zones.
+        """
+        handler = EventHandler(app=self, data=data, context=context)
+        result = handler.run()
+        return result
+
+
+def main(data, context=None, http=None, session=None):
```

Comment:
Wrapping this in a `main` method seems to be an unnecessary level of abstraction and IMO makes your tests less readable with `main.main`.

Instead you should bump this RuntimeState logic directly into the `dns_vm_gc` and test that method.
I also think `main.py` should have a more descriptive name. You could change this to `dns_vm_gc.py`

Reply:
Fixed in ecde344...5eef9da